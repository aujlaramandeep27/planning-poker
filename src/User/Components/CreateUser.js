import loginImg from '../../resources/images/learning_small.jpg';
import React from 'react';
import {connect} from 'react-redux';
import {PostUser} from "../Actions/UserActions";
import {viewMeetings} from "../../Navigation/route-actions";

export function CreateUser(props) {

    return (
        <div className="uk-flex uk-flex-middle uk-height-viewport">
            <div className="uk-width-1-1">
                <img src={loginImg} alt="login"/>
            </div>
            <div className="uk-container uk-width-1-1 uk-margin-small">
                <h1>Welcome to planning poker</h1>
                <h3>Create a new user to get started!</h3>
                <form onSubmit={event => {PostNewUser(event)}}>
                    <div className="uk-margin">
                        <div className="uk-inline uk-width-1-2">
                            <span className="uk-form-icon" data-uk-icon="icon: user"/>
                            <input name="userInputName" className="uk-input" placeholder="Name" type="text"/>
                        </div>
                        <button className="uk-button uk-button-primary uk-width-1-4">
                            CREATE
                        </button>
                    </div>
                </form>
            </div>
        </div>
    );

    function PostNewUser(event) {
        event.preventDefault();
        props.postUser(event.target.userInputName.value.trim());
        props.goToMeetings();
    }

}

const MapDispatchToProps = dispatch => ({
    postUser: (username) => dispatch(PostUser(username)),
    goToMeetings: () => dispatch(viewMeetings()),
});

export default connect(null, MapDispatchToProps)(CreateUser)
