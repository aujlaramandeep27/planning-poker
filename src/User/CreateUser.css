.login-background {
    background-color: #37214d;
}

h1, h2, h3 {
    font-family: 'Frank Ruhl Libre', serif;
    color: #ffffff;
    margin: 2px;
}

.uk-button-primary {
    background-color: #a1dbff;
}