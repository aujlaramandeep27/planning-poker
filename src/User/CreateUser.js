import React from 'react';

export default class CreateUser extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            name: ''
        }
    }
    
    handleChange = (e) => {
        this.setState({
            name: e.target.value
        })
    }

    onSubmit = (e) => {
        e.preventDefault();
        const body = {
         name: this.state.name
        }
        // API call here
        console.log(body);
        this.props.history.push('/meetings');      
     }

    render() {
        return (
            <div className="uk-flex uk-flex-center">
                <div className="uk-card uk-card-default uk-card-body uk-width-1-2@m">
                    <h1 className="uk-heading-small">Create New User</h1>                
                    <input placeholder="Name" className="uk-input uk-margin" 
                        value={this.state.name} 
                        onChange={e => this.handleChange(e)}></input>
                    <button className="uk-button uk-button-primary"
                        onClick={(e) => this.onSubmit(e)}>CREATE</button>
                </div>
            </div>
        );
    }    
}