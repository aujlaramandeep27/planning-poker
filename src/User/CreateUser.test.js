import CreateUser from "./CreateUser";
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const store = mockStore(
    {
        user: {
            items: null,
            error: null,
            loading: false
        }
    });

it('CreateUser renders correctly', () => {
    const createUser = renderer.create(<Provider store={store}><CreateUser/></Provider>).toJSON();
    expect(createUser).toMatchSnapshot();
});