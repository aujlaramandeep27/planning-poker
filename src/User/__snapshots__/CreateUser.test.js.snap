// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUser renders correctly 1`] = `
<div
  className="uk-flex uk-flex-middle uk-height-viewport default-background"
>
  <div
    className="uk-width-1-1"
  >
    <img
      src="learning_small.jpg"
    />
  </div>
  <div
    className="uk-container uk-width-1-1 uk-margin-small"
  >
    <h1>
      Welcome to planning poker
    </h1>
    <h3>
      Create a new user to get started!
    </h3>
    <div
      className="uk-inline uk-width-1-2 uk-margin"
    >
      <span
        className="uk-form-icon"
        uk-icon="icon: user"
      />
      <input
        className="uk-input"
        placeholder="Name"
        type="text"
      />
    </div>
    <Link
      className="uk-button uk-button-primary uk-width-1-4"
      to="/meetings/"
    >
      CREATE
    </button>
  </div>
</div>
`;
